src/script/sign.h

```diff
@@ -223,7 +223,8 @@ struct PSBTInput
         // If there is a non-witness utxo, then don't add the witness one.
         if (non_witness_utxo) {
             SerializeToVector(s, PSBT_IN_NON_WITNESS_UTXO);
-            SerializeToVector(s, non_witness_utxo);
+            OverrideStream<Stream> os(&s, s.GetType(), s.GetVersion() | SERIALIZE_TRANSACTION_NO_WITNESS);
```

Comment:
nit: could use the `WithOrVersion` helper here:
https://github.com/bitcoin/bitcoin/blob/ddc3ec92b0e655a3da21ac2e85ec2e7ecb66c65b/src/streams.h#L66-L70